src/shared/particle_dynamics/diffusion_reaction_dynamics/diffusion_dynamics.hpp

```diff
@@ -159,13 +159,13 @@ DiffusionRelaxationDirichlet<ParticlesType, ContactParticlesType, KernelGradient
 //=================================================================================================//
 template <class ParticlesType, class ContactParticlesType, class KernelGradientType>
 void DiffusionRelaxationDirichlet<ParticlesType, ContactParticlesType, KernelGradientType>::
-    getDiffusionChangeRateDirichletContact(size_t particle_i, size_t particle_j, Vecd &e_ij,
+    getDiffusionChangeRateDirichlet(size_t particle_i, size_t particle_j, Vecd &e_ij,
                                            Real surface_area_ij, const StdVec<StdLargeVec<Real> *> &gradient_species_k)
 {
     for (size_t m = 0; m < this->all_diffusions_.size(); ++m)
     {
         Real diff_coff_ij =
-            this->all_diffusions_[m]->getInterParticleDiffusionCoff(particle_i, particle_j, e_ij);
+            this->all_diffusions_[m]->getInterParticleDiffusionCoff(particle_i, particle_i, Vecd::Zero());
```

Comment:
Probally still use e_ij, because even with boundary this direction will be used for compute the non-isotropic diffusion matrix.

Reply:
I think the non-isotropic diffusion is within the diffusion body, i.e. both particle i and particle j are from the diffusion body. It will use DiffusionInner, rather than particle j being from the contact body.

Reply:
eij has its usage for non-isotropic diffusion.